components/gitpod-protocol/go/gitpod-service.go

```diff
@@ -1730,6 +1730,10 @@ type WorkspaceContext struct {
 	ForceCreateNewWorkspace bool   `json:"forceCreateNewWorkspace,omitempty"`
 	NormalizedContextURL    string `json:"normalizedContextURL,omitempty"`
 	Title                   string `json:"title,omitempty"`
+
+	// Commit context
+	Repository *Repository `json:"repository,omitempty"`
```

Comment:
When is it possible for a workspace to not have a Repository?

Reply:
I'm just following the [protocol definition](https://github.com/gitpod-io/gitpod/blob/01b23af10e7241abc588b1c23bda31bf9b2a9f7c/components/gitpod-protocol/src/protocol.ts), so `WorkspaceContext` interface does not have `Repository` but more specific contexts like `CommitContext` extends from it, I could define all other interfaces but then I think I will need to define a custom json parser so I can check for specific properties and return the correct struct type.
Also If it's expected for `getWorkspace` to always return a context of type `CommitContext` then shouldn't the typing in protocol.ts be updated? what about snapshot context?